working/1426-extension-types/feature-specification.md

```diff
@@ -0,0 +1,948 @@
+whose static type is `E`. It should be noted that the extension method
+named `toString` will only be invoked when the static receiver type is `E`,
+and, e.g., `(o as Object?).toString()` will invoke the implementation of
+`toString()` in the run-time class of `o`.*
+
+*The members of `Object?` are part of the interface of every extension
+type, and this means that it is a compile-time error if said member
+declaration is not a correct override of the declaration in `Object?`. For
+instance, `bool toString(int arg) => true;` is an error.*
+
+*It may not be obvious why we would want to prevent implicit invocations of
+the members of any given extension type. Here is a rationale:*
+
+*Consider the type `int`. This type is likely to be used as the on-type of
+many different extension types, because it allows a very lightweight object
+to play the role as a value with a specific interpretation (say, an `Age`
+in years or a `Width` in pixels). Different extension types are not
+assignable to each other, so we'll offer a certain protection against
+inconsistent interpretations.*
+
+*However, if we have many different extension types with the same or
+overlapping on-types, then it may be impractical to work with: Lots of
+extension methods are applicable to any given expression of that on-type,
+and they are not intended to be used at all, each of them should only be
+used when the associated interpretation is valid, that is, when the static
+type of the receiver is the extension type that declares said member.*
+
+*Hence, we want to support the notion of an extension type whose methods
+are never invoked implicitly. An explicit extension type will do this.*
+
+
+### Allow instance member access using `show` and `hide`
+
+This section specifies the effect of including a non-empty
+`<extensionShowHidePart>` in an extension declaration.
+
+*The show/hide part provides access to a subset of the members of the
+interface of the on-type. For instance, there may be some read-only methods
+that we can safely call on the on-type, because they won't violate any
+invariants associated with the extension type. We could write forwarding
+members in the extension body, but using show/hide can have the same
+effect, and it is much more concise and convenient.*
+
+We use the phrase _extension show/hide part_, or just _show/hide part_ when
+no doubt can arise, to denote a phrase derived from
+`<extensionShowHidePart>`. Similarly, an `<extensionShowClause>` is known
+as an _extension show clause_, and an `<extensionHideClause>` is known as
+an _extension hide clause_, similarly abbreviated to _show clause_ and
+_hide clause_.
+
+The show/hide part specifies which instance members of the on-type are
+available for invocation on a receiver whose type is the given extension
+type.
+
+If the show/hide part is empty, no instance members except the ones
+declared for `Object?` can be invoked on a receiver whose static type is
```

Comment:
Can you hide the `Object?` methods?

Reply:
I would suggest yes. The current wording should include that case, and the discussion section at the end even uses `hide Object?` in an example.